app/models/solidus_stripe/payment_method.rb

```diff
@@ -4,40 +4,133 @@ module SolidusStripe
   class PaymentMethod < ::Spree::PaymentMethod
     preference :api_key, :string
     preference :publishable_key, :string
+    preference :webhook_endpoint_secret, :string, default: ''
 
     validates :available_to_admin, inclusion: { in: [false] }
 
-    def partial_name
-      "stripe"
-    end
+    concerning :Actions do
+      def actions
+        %w[capture void credit]
+      end
 
-    alias cart_partial_name partial_name
-    alias product_page_partial_name partial_name
+      # @param payment [Spree::Payment] the payment we want to know if can be captured
+      # @return [Boolean] true when the payment is in the pending or checkout states
+      def can_capture?(payment)
+        payment.pending? || payment.checkout?
+      end
+
+      # @param payment [Spree::Payment] the payment we want to know if can be voided
+      # @return [Boolean] true when the payment is not failed, voided or invalid
+      def can_void?(payment)
+        payment.can_void? && !payment.completed?
+      end
 
-    def payment_source_class
-      PaymentSource
+      # Indicates whether its possible to credit the payment.  Note that most
+      # gateways require that the payment be settled first which generally
+      # happens within 12-24 hours of the transaction.
+      #
+      # @param payment [Spree::Payment] the payment we want to know if can be credited
+      # @return [Boolean] true when the payment is completed and can be credited
+      def can_credit?(payment)
+        payment.completed? && payment.credit_allowed > 0
+      end
     end
 
-    def gateway_class
-      Gateway
+    concerning :Configuration do
+      def source_required?
+        false
+      end
+
+      def payment_source_class
+        PaymentSource
+      end
+
+      def gateway_class
+        Gateway
+      end
+
+      def partial_name
+        "stripe"
+      end
+
+      alias cart_partial_name partial_name
+      alias product_page_partial_name partial_name
+      alias risky_partial_name partial_name
     end
 
-    def create_profile(payment)
-      payment_intent = payment.source.payment_intent
+    concerning :Order do
+      def find_in_progress_payment_for(order)
+        order.payments
+          .where.not(state: %w[completed invalid void]) # in_progress
+          .find_by(payment_method: self)
+      end
 
-      if payment_intent && payment_intent.customer.blank?
-        payment.payment_method.gateway.request do
-          payment_intent.customer = Stripe::Customer.new email: payment.order.email
-          payment_intent.customer.save
-          payment_intent.save
+      def create_in_progress_payment_for(order)
+        intent = gateway.request do
+          Stripe::PaymentIntent.create({
+            amount: gateway.to_stripe_amount(
+              order.display_total.money.fractional,
+              order.currency,
+            ),
+            currency: order.currency,
+            capture_method: 'manual',
```

Comment:
I understand that here we're telling Stripe that we want to capture payments manually. However, shouldn't we integrate that into Solidus logic for auto-capture?

https://github.com/solidusio/solidus/blob/7be1937c16c9c8bf689010a91e762a833257de87/core/app/models/spree/payment_method.rb#L157

Reply:
Thanks, I set it as a placeholder initially and forgot to connect it to the query method 👍